### ISSUE TITLE
[Question]: Saving 'fine-tuned' Flair embeddings

### ISSUE THREAD

**Issue by msummerfield** [flairNLP/flair#3185]
### Question

I am training a Flair `SequenceTagger` on a particular sequence labelling task using `StackedEmbeddings([FlairEmbeddings('news-forward'), FlairEmbeddings('news-backward')])`.

As I understand it, the weights of the `FlairEmbeddings` are not locked during training, and therefore the embeddings will effectively become 'fine-tuned' on my task. Is this correct?

If so, then I am wondering if, and how, it is possible to save the fine-tuned embeddings to one or more local files, e.g. `my-fwd` and `my-bwd` so that I can use them in other tasks, e.g. by instantiating `StackedEmbeddings([FlairEmbeddings('my-fwd'), FlairEmbeddings('my-bwd')])`.

Can this be done?

Thanks!

**Comment by alanakbik** [comment]
Hello @msummerfield - `FlairEmbeddings` can be fairly easily saved and loaded (see example code below).

However, by default, the `FlairEmbeddings` are not fine-tuned, so they stay the same during training. To make them fine-tuneable, you need to pass `fine_tune=True` when you load the `FlairEmbeddings`:  

```python
# instantiate forward and backwards embeddings with fine_tune=True
forward_lm_embeddings = FlairEmbeddings('news-forward', fine_tune=True)

# save the embedding (you save the underlying language model 'lm')
forward_lm_embeddings.lm.save('tuned-forward-model.pt')

# load it again
loaded_embeddings = FlairEmbeddings('tuned-forward-model.pt', fine_tune=True)
```

We actually never good good results with fine-tuning LSTM-based LMs, but maybe you are more lucky. If so, I'd be curious to hear about your results!

**Comment by msummerfield** [comment]
Hi @alanakbik - thank you for your reply. I had assumed (without checking - my bad!) that as `fine_tune=True` is the default for other embeddings, that it would be the same for `FlairEmbeddings`.

Looking at the code, it appears that I should be able to 'turn on' fine tuning of embeddings in my already trained model as follows:

```
    for emb in model.embeddings.get_named_embeddings_dict().values():
        if isinstance(emb, FlairEmbeddings):
            emb.fine_tune = True
            emb.static_embeddings = False
```

I could then train again, with a smaller initial learning rate, to see whether I can get any additional benefit from fine-tuning.

If I have any luck, I will certainly let you know.

**Comment by alanakbik** [comment]
Thanks! This also raises an important point: We have to add info on the `fine_tune` option to our new docs here: https://flairnlp.github.io/docs/tutorial-embeddings/flair-embeddings - I'm adding a tag to this issue so I don't forget.
